R/transition_risk_profile.R

```diff
@@ -0,0 +1,118 @@
+#' DRAFT of transition risk
+#'
+#' @param products,products_scenarios,scenarios  TODO Data frames.
+#' @param low_threshold,high_threshold  TODO Expressions.
+#'
+#' @return TODO A data frame
+#' @export
+#'
+#' @examples
+#' library(tiltToyData)
+#' library(readr, warn.conflicts = FALSE)
+#' library(dplyr, warn.conflicts = FALSE)
+#'
+#' restore <- options(readr.show_col_types = FALSE)
+#'
+#' products <- read_csv(toy_emissions_profile_products_ecoinvent())
+#' # TODO: Submit a PR to tiltToyData?
+#' products_scenarios <- read_csv(toy_sector_profile_companies()) |>
+#'   select(-c("companies_id", "company_name", "clustered")) |>
+#'   distinct()
+#' scenarios <- read_csv(toy_sector_profile_any_scenarios())
+#'
+#' transition_risk_profile(products, products_scenarios, scenarios)
+#'
+#' options(restore)
+transition_risk_profile <- function(products,
+                                    # TODO: Do we need a new dataset in tiltToyData?
+                                    products_scenarios,
+                                    scenarios,
+                                    low_threshold = tercile(.data$transition_risk_score, 1),
+                                    high_threshold = tercile(.data$transition_risk_score, 2)) {
+  score_transition_risk(products, products_scenarios, scenarios) |>
+    # Add thresholds
+    mutate(
+      transition_risk_low_threshold = !!rlang::enquo(low_threshold),
+      transition_risk_high_threshold = !!rlang::enquo(high_threshold),
+      .by = "benchmark_tr_score"
+    ) |>
+    # ... do something else?
```

Comment:
Here is the code to understand how low and high threshold arguments will look like in transition risk indicator function (user-facing)